### PR TITLE
Migrate header check to setupcheck API

### PR DESCRIPTION
**Pull request by come-nc**
See #41364 

## Summary

Security related header check migrated to new API and done from backend now.
TODO:
- [x] Migrate the tests
- [x] Merge the HSTS check in there

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/lib/SetupChecks/OcxProviders.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function run(): SetupResult {
 		];
 
 		foreach ($providers as $provider) {
-			foreach ($this->runHEAD($this->urlGenerator->getWebroot() . $provider) as $response) {
```
Comment (susnux) [inline apps/settings/lib/SetupChecks/OcxProviders.php:71]: If that is removed, then drop also `runHEAD` function?

Comment (come-nc) [reply apps/settings/lib/SetupChecks/OcxProviders.php:71]: Still used in other checks I think but yeah we could move all of them.

File: apps/settings/lib/SetupChecks/SecurityHeaders.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+						'link' => 'https://www.w3.org/TR/referrer-policy/',
+					];
+				}
+
+				$transportSecurityValidity = $response->getHeader('Strict-Transport-Security');
+				$minimumSeconds = 15552000;
+				if (preg_match('/^max-age=(\d+)(;.*)?$/', $transportSecurityValidity, $m)) {
+					$transportSecurityValidity = (int)$m[1];
+					if ($transportSecurityValidity < $minimumSeconds) {
+						$msg .= $this->l10n->t('- The `Strict-Transport-Security` HTTP header is not set to at least `%d` seconds (current value: `%d`). For enhanced security, it is recommended to enable HSTS.', [$minimumSeconds, $transportSecurityValidity])."\n";
```
Comment (Altahrim) [inline apps/settings/lib/SetupChecks/SecurityHeaders.php:122]:
```suggestion
						$msg .= $this->l10n->t('- The `Strict-Transport-Security` HTTP header is not set to at least `%d` seconds (current value: `%d`). For enhanced security, it is recommended to use a long HSTS policy.', [$minimumSeconds, $transportSecurityValidity])."\n";
```

**Review by Altahrim — APPROVED** [review]
If domain contains only one dot (`xxxxx.yyy`), we could also advise to use HSTS `includeSubDomains` and `preload` (https://hstspreload.org/)

**Review by susnux — APPROVED** [review]

**Comment by come-nc** [comment]
> If domain contains only one dot (`xxxxx.yyy`), we could also advise to use HSTS `includeSubDomains` and `preload` (https://hstspreload.org/)

I do not have any knowledge about this, I’d say that would be a followup, let’s migrate the feature first.

**Pull request merged**